tools/Update-Progress.ps1

```diff
@@ -0,0 +1,36 @@
+# Update-Progress.ps1
+
+function Update-Progress {
+    param (
+        [Parameter(Mandatory, Position = 0)]
+        [ValidateNotNullOrEmpty()]
+        [string]$StatusMessage,
+
+        [Parameter(Mandatory, Position = 1)]
+        [ValidateRange(0, 100)]
+        [int]$Percent,
+
+        [Parameter(Position = 2)]
+        [string]$Activity,
+
+        [Parameter(Position = 3)]
+        [switch]$LogProgress
+    )
+
+    # Default activity to "Processing" if not provided
+    if (-not $Activity) {
+        $Activity = "Processing"
+    }
```

Comment:
```suggestion
        [Parameter(Position = 2)]
        [string]$Activity = "Processing",  # Default activity to "Processing" if not provided

        [Parameter(Position = 3)]
        [switch]$LogProgress
    )
```
You can assign a parameter a default value, which'll be used if no value was passed when function is called. It's a standard feature in PowerShell, and IMO it makes the code shorter and more readable. plus I've left the comment to make it super clear that the default value will be `Processing`.